core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/ThrottlingHandler.java

```diff
@@ -0,0 +1,137 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.Header;
+
+import com.google.common.base.Strings;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicy;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+/**
+ * This class implements the retry logic for throttled requests. HTTP status
+ * code 429 and 439.
+ * 
+ * @author jamdavi
+ *
+ */
+public class ThrottlingHandler implements TransmissionHandler {
+
+	private TransmissionPolicyManager transmissionPolicyManager;
+	private final static String RESPONSE_RETRY_AFTER_DATE_FORMAT = "E, dd MMM yyyy HH:mm:ss";
+
+	/**
+	 * Ctor
+	 * 
+	 * Constructs the ThrottlingHandler object.
+	 * 
+	 * @param policy
+	 *            The {@link TransmissionPolicyManager} object that is needed to
+	 *            control the back off policy.
+	 */
+	public ThrottlingHandler(TransmissionPolicyManager policy) {
+		this.transmissionPolicyManager = policy;
+	}
+
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+	}
+
+	/**
+	 * Provides the core logic for the retransmission
+	 * 
+	 * @param args
+	 *            The {@link TransmissionHandlerArgs} for this transmission.
+	 * @return Returns a pass/fail for handling this transmission.
+	 */
+	boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getRetryHeader() != null && args.getTransmission() != null
+				&& args.getTransmissionDispatcher() != null) {
+			args.getTransmission().incrementNumberOfSends();
```

Comment:
I see that PartialResponseHandler class doesn't increment the number of sends like this. Is there a reason why it is omitted there? Because eventually we are retrying correct?

Reply:
In this instance we're creating a brand new transmission to send based on the result of the old one. We are not going to dispatch the entire transmission so no need to increment.

Reply:
ok